server/plugin/plugin.go

```diff
@@ -56,6 +59,52 @@ func (p *Plugin) OnDeactivate() error {
 	return nil
 }
 
+func (p *Plugin) closeBackgroundJob() {
+	if err := p.backgroundJob.Close(); err != nil {
+		p.API.LogError("Failed to close background job", "Error", err.Error())
+	}
+}
+
+func (p *Plugin) deactivateJob() {
+	if p.backgroundJob != nil {
+		p.closeBackgroundJob()
+		if err := p.API.KVDelete(cronPrefix + PublishSeriveNowVAIsTypingJobName); err != nil {
```

Comment:
No need for the cron prefix. Instead, you should hash the key. See https://github.com/Brightscout/mattermost-plugin-hamilton-artifactory/blob/063b96222d3fcbc16506964b31a8ce95b6618e30/server/plugin.go#L91